slick/src/main/scala/slick/jdbc/MySQLProfile.scala

```diff
@@ -94,6 +94,12 @@ trait MySQLProfile extends JdbcProfile { profile =>
         else l
       }
     }
+
+    //Reference: https://github.com/slick/slick/issues/1419
+    override def createTableNamer(meta: MTable): TableNamer = new TableNamer(meta){
+      override def schema = meta.name.catalog
+      override def catalog = meta.name.schema 
```

Comment:
I haven't done much analysis on this so I'm not sure what should be the right decision here. The `catalog` field could be set to `None` or remain unchanged, or we could choose to be consistent with the definitions of the MySQL JDBC driver and leave nothing changed. They all make sense to me. It depends on what is the expected behavior, and how this would affect other parts in the source code generator.